### PR TITLE
Post about ad scams

### PR DESCRIPTION
**Pull request by davidfischer**
It's a bit of a balance between too little and too much details but hopefully I struck the right balance.

## Screenshot

Screenshot 2022-09-13 at 14-37-49 Stopping Ad Fraud Brazen Scams and How We Keep Our Network Safe

### CODE REVIEW

File: content/posts/2022-stopping-ad-fraud.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
+Title: Stopping Ad Fraud: Brazen Scams and How We Keep Our Network Safe
+Date: September 20, 2022
+description: The advertising business has a well-deserved reputation for scams and fraud. We detail some of what we've seen and how we keep our network safe.
+tags: security, adtech
+authors: David Fischer
+image: /images/posts/2022-stopping-ad-fraud.jpg
+image_credit: <span>Photo by <a href="https://unsplash.com/@jdent?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Jason Dent</a> on <a href="https://unsplash.com/s/photos/security?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Unsplash</a></span>
+
+
+As we mentioned in our [regular updates recently]({filename}newsletter-july-2022.md),
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:10]:
```suggestion
As we mentioned in our [newsletter recently]({filename}newsletter-july-2022.md),
```

File: content/posts/2022-stopping-ad-fraud.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+As our network grows, we knew we would have a target on our back for scammers.
+Since we don't track users and have pledges around what data we retain
+in our [Do Not Track policy](https://server.ethicalads.io/.well-known/dnt-policy.txt),
+scammers will go after us and think we're easy pickings.
+When there's money to be made online, people will try to do it.
+
+The way these scams usually work is that a person will set up a site, request an ad publisher account,
+and then simulate fake traffic to the site and rake in the earnings.
+I'm sure folks working at Google have seen thousands of variations on this
+and they're probably pretty good at defending against it.
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:28]: Not sure what this line is accomplishing? Seems like we want to say something different here.

File: content/posts/2022-stopping-ad-fraud.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+
+## Our defenses
+
+Like most ad networks, we have a number of defenses to automatically try to detect and prevent this.
+While our code is open source, some of these defenses rely on configuration outside of our open source code.
+Just to talk to some of them:
+
+- We use both free and paid databases of blocklists belonging to known VPNs or known automated traffic.
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:37]: Remove 2x known. Are data centers & known automated traffic hosts the same?

```suggestion
- We use both free and paid databases of blocklists belonging to known VPNs, data centers, and automated traffic hosts.
```

File: content/posts/2022-stopping-ad-fraud.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+- Our upstream CDN provider, Cloudflare, can block or challenge problematic clients,
+  but we also use the threat score metric they provide.
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:39]:
```suggestion
- Our upstream CDN provider, Cloudflare, can block problematic clients,
  and we also use the threat score metric they provide.
```

File: content/posts/2022-stopping-ad-fraud.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+- We do a number of other normal protections that ad networks do
+  like use nonces to prevent ads being viewed multiple times
+  and making sure the client requesting an ad makes sense.
+
+These defenses work very well at making sure we're counting real traffic for our advertisers
+and paying our publishers appropriately and fairly.
+We routinely hear from our advertisers that their own analytics shows *more traffic* coming from us
+than our reporting.
+That's not to say they're wrong and we're right but just that we may not be counting exactly the same thing.
+Analytics programs are interested in all non-bot traffic while we report only billable ad traffic.
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:51]: This is probably worth highlighting more. Perhaps even in a pull quote, if that's a thing we can do?

File: content/posts/2022-stopping-ad-fraud.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+
+Probably our biggest defense, however, is our relative size to the big networks.
+Unlike larger ad networks, we vet every single publisher site we onboard.
+We talk to every publisher, we only do a few at most in any given week, and the process isn't automated.
+A person from our team looks at the site where EthicalAds will appear both before our ads appear and after they appear but before paid ads are turned on to make sure everything looks good.
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:56]: 3x appear! :D 

```suggestion
A person from our team looks at our publisher sites both before and after paid ads are turned on to make sure everything looks good.
```

File: content/posts/2022-stopping-ad-fraud.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+In late July, we got some requests that appeared to come from documentation for decent sized open source libraries.
+Docs sites for code modules are a natural place to put an ad in front of an engaged audience
+and it puts money directly into the hands of open source maintainers.
+This is our bread and butter and we built EthicalAds to do exactly this for [Read the Docs](https://readthedocs.org/).
+
+When we first let one of these sites onto our network (they'd switched from a competitor),
+we knew something was wrong almost immediately.
+There was a little more traffic than was normal for a library like this, but a huge percentage looked automated.
+We regularly look at a few internal metrics like what percentage of ads we send are viewed and how long they appear on the screen. But the metrics screamed that this new site wasn't normal traffic at all.
+Within a few days, we kicked them off and refunded the small amount of traffic that got through our checks.
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:73]:
```suggestion
Within a few days, we kicked them off and refunded the small amount of traffic that got through to our advertisers.
```

File: content/posts/2022-stopping-ad-fraud.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+made a site that proxied traffic to the real library's site but could inject ads,
+and created fake personas to impersonate the real library's devs:
+names and emails that loosely matched one of the developers on GitHub.
+There were dozens of examples but to name some specific modules (these are the real ones):
+
+- [AsyncJS](https://github.com/caolan/async)
+- [NobloxJS](https://noblox.js.org/)
+- [DatafluxJS](https://github.com/massimocandela/dataflux)
+- [Tool Cool Color Picker](https://github.com/toolcool-org/toolcool-color-picker)
+
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:86]: Probably want to link to archive.org or similar if we can, or just assume these will go away.

File: content/posts/2022-stopping-ad-fraud.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+## Keeping our defenses strong
+
+In response to knowing we're now on the scammers' radar,
+we made a few changes to make sure our network stays safe:
+
+- We do some additional scrutiny on the clients viewing our ads.
+- Our vetting process for incoming publishers has become more formalized
+  and got a few additional steps to ensure sites on our network are a good fit.
```
Comment (ericholscher) [inline content/posts/2022-stopping-ad-fraud.md:102]:
```suggestion
  and got a few additional steps to ensure sites on our network look real and on topic.
```

**Review by ericholscher — APPROVED** [review]
This is great. I do think that if we wanted this to get on HN we would focus more on the impersonation, and less on our defenses. I'd expect a title like `Watch out: ad scammers are impersonating open source developers.

I'd actually propose perhaps turning this into 2 different posts that we publish at the same time, one that highlights our protections and one that highlights the fraud which links to our post on additional measures.

I think there's a ~33% the ad fraud information gets on HN front page, but probably 5% that this post as written does.

File: content/posts/2022-stopping-ad-fraud.md (hunk 8 above)
Comment (davidfischer) [reply content/posts/2022-stopping-ad-fraud.md:86]: These are the "real" sites, not the clones. Do you think we should link to the clones?

**Comment by davidfischer** [comment]
> This is great. I do think that if we wanted this to get on HN we would focus more on the impersonation, and less on our defenses. I'd expect a title like `Watch out: ad scammers are impersonating open source developers and placing ads on forks of their sites`.
> 
> I'd actually propose perhaps turning this into 2 different posts that we publish at the same time, one that highlights our protections and one that highlights the fraud which links to our post on additional measures.
> 
> I think there's a ~33% the ad fraud information gets on HN front page, but probably 5% that this post as written does, because it's so focused on our product and not the problem itself.

I tend to agree with your assessment except for turning this into 2 posts. Maybe our defenses just aren't as important to highlight. I don't think we want to have two simultaneous blogs. Maybe just highlight the scams more and talk about our defenses less or at the end.

**Comment by ericholscher** [comment]
I think the defenses could also go into a page on our site about ad fraud, if we wanted. I don't think posting 2 blogs is inherently bad though 🤷

**Comment by davidfischer** [comment]
I updated the post to focus more on the impersonating and less on our defenses. I named the specific mirrors in question although those will probably go down at some point.

**Review by ericholscher — APPROVED** [review]
Looks good. I do wonder if we should keep that content around our defenses somewhere on the website. It seems valuable, but also understand if you don't want to give too much away…

**Pull request merged**